lib/client/clientcache/clientcache.go

```diff
@@ -45,8 +45,20 @@ type Cache struct {
 type clientWithCert struct {
 	// client is cluster client.
 	client *client.ClusterClient
-	// cert used in TeleportClient.ConnectToCluster to create the client
+	// coreTLSCert is the cert used in TeleportClient.ConnectToCluster to create the client.
 	coreTLSCert []byte
+	// readCoreTLSCert reads a fresh cert from disk.
+	readCoreTLSCert func() ([]byte, error)
```

Comment:
I don't think it's very Go-like, I assume the more idiomatic way to do it is to create a profile interface with `TLSCert` method.